docs/blog/posts/sled-integration.md

```diff
@@ -0,0 +1,248 @@
+structures which were actually useful for what bpfman. More
+specifically, data would come in via a GRPC server as specified
+in `bpfman.v1.rs` where rust types are tightly coupled to the protobuf
+definition. In `rpc.rs` data was then converted to an internal set of structures
+defined in `command.rs`.  Prior to [pull request #683](https://github.com/bpfman/bpfman/pull/683)
+there was an explosion of types, with each bpfman command having it's own set of
+internal structures and enums.  Now, most of the data for a program that bpfman
+needs internally for all commands to manage an eBPF program is stored in
+the `ProgramData` structure which we'll take a deeper look at a bit later.
+Additionally, there is much more complexity introduced for XDP and TC program types
+which rely on extra eBPF dispatcher programs in order to prioritize multiple programs
+for a single network interface, however this article will try to instead focus
+on the simpler examples.
+
+Aside from the complexity of the data organization itself which will be further
+simplified as the grpc server is deprecated in favor of packaging, bpfman has to be
+persistent across restarts. In order to make this work raw data was often
+flushed to disk in the form of JSON files (all types in `command.rs` needed to
+implement [serde's](https://serde.rs/) `Serialize` and `Deserialize`). Additionally, specific
+significance would also be encoded to bpfman's directory structure, i.e all program
+related information was encoded in `/run/bpfd/programs/<ID>`. The extra infrastructure
+and failure modes introduced by this process was a constant headache, pushing the
+community to find a better solution.
+
+## Why Sled?
+
+[Sled](https://github.com/spacejam/sled) is an open source project described in
+github as "the champagne of beta embedded databases".  The "reasons" for choosing
+an embedded database from the [project website][sled] are pretty much spot on:
+
+```yaml
+Embedded databases are useful in several cases:
+
+- you want to store data on disk, without facing the complexity of files
+- you want to be simple, without operating an external database
+- you want to be fast, without paying network costs
+- using disk storage as a building block in your system
+```
+
+As discussed in the previous section, persistence across restarts, is one of
+bpfman's core design constraints, and with sled we *almost* get it for free!
+Additionally due to the pervasive nature of data management to bpfman's core
+workflow the data-store needed to be kept as simple and light weight as possible,
+ruling out heavier production-ready external database systems such as MySQL or
+Redis.
+
+Now this mostly focused on why embedded dbs in general, but why did we choose
+sled.....well because it's written in :crab: Rust :crab: of course! Apart from the
+obvious we took a small dive into the project before rewriting everything by
+[transitioning the OCI bytecode image library](https://github.com/bpfman/bpfman/pull/861)
```

Comment:
nit: Prefer adding the link at the end of the section
i.e use `[transitioning the OCI bytecode image library]` here and 

```md
[transitioning the OCI bytecode image library]: https://github.com/bpfman/bpfman/pull/861
```

at the end of this block.